server/auth.go

```diff
@@ -92,11 +94,29 @@ func (s *Server) registerAuthRoutes(g *echo.Group) {
 					return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("vcs do not exist, name: %v, ID: %v", gitlabLogin.Name, gitlabLogin.Name)).SetInternal(err)
 				}
 
+				// exchange Oauth Token
+				oauthToken, err := vcsPlugin.Get(vcsFound.Type, vcsPlugin.ProviderConfig{Logger: s.l}).ExchangeOauthContent(
+					ctx,
+					vcsFound.InstanceURL,
+					common.OauthContext{
+						ClientID:     vcsFound.ApplicationID,
+						ClientSecret: vcsFound.Secret,
+					},
+					gitlabLogin.Code,
+					fmt.Sprintf("%s:%d/oauth/callback", s.frontendHost, s.frontendPort),
+				)
+				if err != nil {
+					return echo.NewHTTPError(http.StatusInternalServerError, "Failed to exchange OAuth token").SetInternal(err)
+				}
+				if oauthToken == nil {
```

Comment:
Since we define the interface, we can set the contract that if the err is nil, then the OAuthToken should be non-nil.

Thus the client code does not need to handle this.

Reply:
Yeah, we can avoid using oauthToken as a pointer here.